packages/node_modules/express-pouchdb/lib/index.js

```diff
@@ -185,6 +185,38 @@ module.exports = function (startPouchDB, opts) {
     next();
   });
 
+  app.use(function (req, res, next) {
+    var regexFindDbName = /\/([^\/]*)/
+    var dbName = decodeURIComponent(req.url.match(regexFindDbName)[1])
+
+    if (!dbName) {
+      return next();
+    }
+
+    // check for internal routes GET /_all_dbs
+    if (dbName[0] === '_') {
+      for (var i = 0, route; i < app._router.stack.length; i++) {
+        route = app._router.stack[i].route;
+        if (route) for (var j = 0, method; j < route.stack.length; j++) {
+          method = route.stack[j].method.toUpperCase()
+          if (method === req.method && route.path.substr(1, dbName.length) === dbName) {
```

Comment:
Yes, I think you do if you go this way. You're checking if the **current** method is the same as the method of the **route** you're trying to match. If you don't check, it would allow you to do things like POST /_all_dbs (which is nonsensical).